cobigen/cobigen-openapiplugin-parent/cobigen-openapiplugin/src/main/java/com/devonfw/cobigen/openapiplugin/inputreader/OpenAPIInputReader.java

```diff
@@ -413,33 +415,53 @@ private PropertyDef extractReferenceProperty(OpenApi3 openApi, Schema componentS
      * @return list of {@link PathDef}'s
      */
     private List<PathDef> extractPaths(Map<String, ? extends Path> paths, String componentName) {
+        Matcher matcher;
+        Pattern pattern;
+        String match, root_component = null, version = null;
+
         List<PathDef> pathDefs = new LinkedList<>();
+
         for (String pathKey : paths.keySet()) {
             if (pathKey.toLowerCase().contains(componentName.toLowerCase())) {
                 String[] mp = pathKey.split("/");
                 String pathUri = "/";
-                for (int i = 3; i < mp.length; i++) {
-                    pathUri = pathUri.concat(mp[i] + "/");
+
+                try {
+                    match = "^\\/[^\\/]+\\/+[^\\/]+\\/(.+)";
+                    pattern = Pattern.compile(match);
+                    matcher = pattern.matcher(pathKey);
+                    matcher.find();
+                    pathUri += matcher.group(1);
```

Comment:
There is a better way to implement this, you can check first whether this group contains at least 2 groups:

```java
if (matcher.groupCount() > 1) {
                    pathUri += matcher.group(1);
                    if (pathUri.substring(pathUri.length() - 1) != "/") {
                        pathUri += "/";
                    }
                }
```

In this case, you don't need to throw any exception.

Reply:
Actually, I m only extracting one group, which is `(.+)`, the part that comes after **rootComponenet/version/thisPartIsExracted**. If the given path is too short <-> does not match the written match, the exception occurs.
In this case, `matcher.groupCount()` would never be > 1.

Reply:
After discussing this, we found out that `groupCount()` shouldn't be used. We will go for other approaches like `if (matcher.matches())` or `if (matcher.find())`